components/BlogPosts.vue

```
@@ -4,57 +4,71 @@ const {
 }>()
 
 const { transitionClasses } = useAnimations()
 
-const postsResult = await fromPromise(
-  (async () => {
-    let query = queryCollection('blog')
+const displayPosts = computed(() => {
+  if (type === 'custom' && posts) {
+    return posts
+  }
+  return postsData.value
```

ESLint

[error] 'postsData' was used before it was defined.

(ts/no-use-before-define)